angular/projects/spark-angular/src/lib/components/sprk-masthead/sprk-masthead.component.ts

```diff
@@ -588,89 +270,69 @@ export class SprkMastheadComponent implements AfterContentInit {
       });
     }
 
-    if (this.isNarrowNavOpen) {
-      classArray.push('sprk-c-Masthead--open');
-    }
-
-    if (this.isScrolled) {
-      classArray.push('sprk-c-Masthead--scroll');
-    }
-
-    if (this.isHidden) {
-      classArray.push('sprk-c-Masthead--hidden');
+    if (
+      this.collapsibleNavDirective &&
+      !this.collapsibleNavDirective.isCollapsed &&
+      this.isNarrowViewport
+    ) {
+      classArray.push('sprk-c-Masthead--is-open');
     }
 
-    return classArray.join(' ');
-  }
-
-  /**
-   * @ignore
-   */
-  getNarrowNavClasses(): string {
-    const classArray: string[] = [];
-
-    if (this.additionalNarrowNavClasses) {
-      this.additionalNarrowNavClasses.split(' ').forEach((className) => {
-        classArray.push(className);
-      });
+    if (this.isPageScrolled) {
+      classArray.push('sprk-c-Masthead--is-scrolled');
     }
 
-    return classArray.join(' ');
-  }
-
-  /**
-   * @ignore
-   */
-  getSecondaryNavClasses(): string {
-    const classArray: string[] = [
-      'sprk-c-Masthead__big-nav-items',
-      'sprk-o-Stack',
-      'sprk-o-Stack--misc-a',
-      'sprk-o-Stack--center-row',
-      'sprk-o-Stack--split@xxs',
-      'sprk-b-List',
-      'sprk-b-List--bare',
-    ];
-
-    if (this.additionalBigNavClasses) {
-      this.additionalBigNavClasses.split(' ').forEach((className) => {
-        classArray.push(className);
-      });
+    if (this.isMastheadHidden && this.isNarrowViewport) {
```

Comment:
Before we had a bug where we were hiding the navigation on large viewports when the user scrolled up. We want the masthead to be sticky in large viewports. This check for if its a narrowviewport fixes that.